src/entities/activity/model/mappers.ts

```diff
@@ -64,13 +63,82 @@ function mapConditionalLogic(dto: ConditionalLogicDto | null) {
         conditionalLogic: {
           match: dto.match,
           conditions: dto.conditions.map(condition => {
+            const { itemName, ...rest } = condition;
             const updatedCondition = {
-              ...condition,
-              activityItemName: condition.itemName,
+              ...rest,
+              activityItemName: itemName,
             };
-
             // @ts-expect-error
             delete updatedCondition.itemName;
+            switch (condition.type) {
```

Comment:
Some cases are missing from this one, that aren't straightforward

```ts
payload: {
  value: number;
};
```

General conditions
case 'BETWEEN':
case 'OUTSIDE_OF':

Row option conditions
case 'EQUAL_TO_ROW_OPTION':
case 'NOT_EQUAL_TO_ROW_OPTION':

Slider row conditions
case 'GREATER_THAN_SLIDER_ROWS':
case 'LESS_THAN_SLIDER_ROWS':
case 'EQUAL_TO_SLIDER_ROWS':
case 'NOT_EQUAL_TO_SLIDER_ROWS':
case 'BETWEEN_SLIDER_ROWS':
case 'OUTSIDE_OF_SLIDER_ROWS':

Should those be added?

Reply:
not really, thats not necessary, but it would be good, actually no one of I added is necessary as long I have "updatedCondition.payload = condition.payload;", but I really think this mapper that I created helps to understand how the payload is working, and being applied, so I will add for those that is missing.

Reply:
the only ones I dont recommend to be mapped, right now is BETWEEN, OUSIDE_OF, GREATER_THAN ..., and the a couple more, because its being used for multiples parts and it must to be generic, or we could map but keep it generic with same payload of "default"

Reply:
done I added a case for those ones that are specific, like GREATER_THAN_SLIDER_ROWS ...